.github/workflows/tests.yml

```diff
@@ -0,0 +1,40 @@
+name: Go Tests
+on:
+  push:
```

Comment:
Can you also add the tests for every PR?
Also would be nice to integrate Dependabot with this repo for updating the dependencies regularly.

Reply:
Added tests for all the PR and a dependabot to check for gomod file